hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -474,16 +482,37 @@ onOpenNetworkReqSn env ledger st otherParty sv sn requestedTxIds mDecommitTx =
   requireApplicableDecommitTx cont =
     case mDecommitTx of
       Nothing -> cont (confirmedUTxO, Nothing)
-      Just decommitTx ->
-        -- Spec: require ̅S.𝑈 ◦ txω /= ⊥
-        case applyTransactions ledger currentSlot confirmedUTxO [decommitTx] of
-          Left (_, err) ->
-            Error $ RequireFailed $ SnapshotDoesNotApply sn (txId decommitTx) err
-          Right newConfirmedUTxO -> do
-            -- Spec: 𝑈_active ← ̅S.𝑈 ◦ txω \ outputs(txω)
-            let utxoToDecommit = utxoFromTx decommitTx
-            let activeUTxO = newConfirmedUTxO `withoutUTxO` utxoToDecommit
-            cont (activeUTxO, Just utxoToDecommit)
+      Just decommitTx
+        -- Spec: if v = S̄.v
+        | sv == confVersion ->
+            case confUTxOToDecommit of
+              Nothing ->
+                -- Spec: require ̅S.𝑈 ◦ txω /= ⊥
+                case applyTransactions ledger currentSlot confirmedUTxO [decommitTx] of
+                  Left (_, err) ->
+                    Error $ RequireFailed $ SnapshotDoesNotApply sn (txId decommitTx) err
+                  Right newConfirmedUTxO -> do
+                    -- Spec: 𝑈_active ← ̅S.𝑈 ◦ txω \ outputs(txω)
+                    let utxoToDecommit = utxoFromTx decommitTx
+                    let activeUTxO = newConfirmedUTxO `withoutUTxO` utxoToDecommit
+                    cont (activeUTxO, Just utxoToDecommit)
```

Comment:
Could: align this closer to what we have in the spec, i.e. make this block work with `mDecommitTx`

That way, we would have the same conditionals. However, I think it is semantically equivalent to what we have in the spec (although hard to see).